pina/trainer.py

```diff
@@ -0,0 +1,29 @@
+""" Solver module. """
+
+import lightning.pytorch as pl
+from .utils import check_consistency
+from .dataset import DummyLoader
+from .solver import SolverInterface
+
+class Trainer(pl.Trainer):
```

Comment:
Why do we keep the solver and the trainer in two different classes?

Reply:
I think is the best way to create multiple solvers, and avoid every time to define the same line of code. Of course you could define the trainer in `SolverInterface` class, so that every solver inherits it. However, if one wants to define its own trainer (e.g. by inheriting from lightining) and use it for a specific solver (e.g. PINN, XPINN, FNO, ...), it would be not possible to do. 

Also semantically, I think is better to separate all the part where we train (e.g. gpu/cpu, pruning, ...) to the solver part. We can discuss it further if you want.

Reply:
Oh ok, now I see it. Yes, I agree, better design!
The only thing I would change is the `Network` class, which is nice to keep in, even if just empty at the moment.
Also maybe `Solver.model` to a dictionary of models `Solver.models`?